src/node_report.cc

```diff
@@ -857,9 +857,11 @@ std::string TriggerNodeReport(Isolate* isolate,
   // Determine the required report filename. In order of priority:
   //   1) supplied on API 2) configured on startup 3) default generated
   if (!name.empty()) {
-    THROW_IF_INSUFFICIENT_PERMISSIONS(
-        env, permission::PermissionScope::kFileSystemWrite, name, name);
-    // Filename was specified as API parameter.
+    if (env != nullptr) {
```

Comment:
I think we should add an assertion here to make sure fs permissions are checked. For the case where env is nullptr, we might have a bug.

Reply:
@anonrig is that the same as the case for the lines a bit lower where a similar null check is used or is this case different?